### ISSUE TITLE
`assertObjectNotEquals()`

### ISSUE THREAD

**Issue by lstrojny** [sebastianbergmann/phpunit#5811]
`assertObjectEquals` is great for VOs and it would be lovely to have the opposite available as well, `assertObjectNotEquals`. Sure, it’s trivial to polyfill but still it would be lovely if PHPUnit could offer it out of the box.

```php
public static function assertObjectNotEquals(object $expected, object $actual, string $method = 'equals', string $message = ''): void
{
    self::assertThat(
        $actual,
        self::logicalNot(self::objectEquals($expected, $method)),
        $message,
    );
}
```

**Comment by sebastianbergmann** [comment]
Back when `assertObjectEquals()` was implemented, I wrote in #4467

> An inverse of `assertObjectEquals()` does not make sense to us and will not be implemented.

I wrote the above because I was not convinced that could be inversed as easily as you propose. Of course, I am happy to be proven wrong. :)

Quoting again from #4467:

> * A method with name `$method` must exist on the `$actual` object
> * The method must accept exactly one argument
> * The respective parameter must have a declared type
> * The `$expected` object must be compatible with this declared type
> * The method must have a declared `bool` return type
> 
> If any of the aforementioned assumptions is not fulfilled or if `$actual->$method($expected)` returns `false` then the assertion fails.

An exception is raised if any of the assumptions is not met.

Given the above, do you still think that "just" using `self::logicalNot(self::objectEquals($expected, $method))` instead of `self::objectEquals($expected, $method)` is good enough to achieve what you desire?

**Comment by lstrojny** [comment]
Sure, that’s fine. The preconditions stay the same and error if unmet and if the equals method returns false, an assertion failure is recorded.
